fish-rust/src/common.rs

```diff
@@ -89,5 +89,5 @@ pub fn escape_string(s: &wstr, style: EscapeStringStyle) -> WString {
         EscapeStringStyle::Regex => ffi::escape_string_style_t::STRING_STYLE_REGEX,
     };
 
-    ffi::escape_string(s.as_ptr(), flags_int.into(), style).from_ffi()
+    ffi::escape_string_len(s.as_ptr(), s.len(), flags_int.into(), style).from_ffi()
```

Comment:
Ouch, that one's on me, but it should already be fixed on master: 7bab4c4ddabd1ce2c4dace55f7215e47e9665736